### ISSUE TITLE
Add test for `load_registry_from_doi` when the doi has a slash in the filename

### ISSUE THREAD

**Issue by santisoler** [fatiando/pooch#341]
**Description of the desired feature:**

We should add a test function that checks if the `load_registry_from_doi` method introduced in #325 will work with the bugfix introduced in #337. Probably would be better to create a utils function that returns the url to the special Zenodo repo that contains a slash in the filename and reuse it among the test functions.

This could be solve in #337 once #325 is merged. Otherwise, it's ok to leave it for its own PR.

**Are you willing to help implement and maintain this feature?**  For sure!
